common/metrics.py

```
@@ -0,0 +1,223 @@
+            if hits is None or misses is None:
+                return 0.0
+
+            hits_value = getattr(hits, "_value", 0)
+            misses_value = getattr(misses, "_value", 0)
+            total = hits_value + misses_value
+
+            if total == 0:
+                return 0.0
+
+            return (hits_value / total) * 100.0
+        except Exception:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)